tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py

```diff
@@ -191,36 +209,230 @@ def test_delete_registered_model_tag_unsupported(store):
+                "POST",
+                message_to_json(
+                    GenerateTemporaryModelVersionCredentialsRequest(
+                        name=name, version=version, operation=MODEL_VERSION_READ_WRITE
+                    )
+                ),
+            ): model_version_temp_credentials_response,
+            (
+                "/api/2.0/mlflow/unity-catalog/model-versions/finalize",
+                "POST",
+                message_to_json(FinalizeModelVersionRequest(name=name, version=version)),
+            ): FinalizeModelVersionResponse(),
+        }
+        response_message = req_info_to_response[
+            (endpoint, method, json.dumps(kwargs["json"], indent=2))
+        ]
+        mock_resp = mock.MagicMock(autospec=Response)
+        mock_resp.status_code = 200
+        mock_resp.text = message_to_json(response_message)
+        return mock_resp
+
+    return request_mock
+
+
+def _assert_create_model_version_endpoints_called(
+    request_mock, name, source, version, run_id=None, description=None
+):
+    """
+    Asserts that endpoints related to the model version creation flow were called on the provided
+    `request_mock`
+    """
+    for endpoint, proto_message in [
+        (
+            "model-versions/create",
+            CreateModelVersionRequest(
+                name=name, source=source, run_id=run_id, description=description
+            ),
+        ),
+        (
+            "model-versions/generate-temporary-credentials",
+            GenerateTemporaryModelVersionCredentialsRequest(
+                name=name, version=version, operation=MODEL_VERSION_READ_WRITE
+            ),
+        ),
+        (
+            "model-versions/finalize",
+            FinalizeModelVersionRequest(name=name, version=version),
+        ),
+    ]:
+        _verify_requests(
+            http_request=request_mock,
+            endpoint=endpoint,
+            method="POST",
+            proto_message=proto_message,
+        )
+
+
+def test_create_model_version_aws(store, tmp_path):
+    access_key_id = "fake-key"
+    secret_access_key = "secret-key"
+    session_token = "session-token"
+    aws_temp_creds = TemporaryCredentials(
+        aws_temp_credentials=AwsCredentials(
+            access_key_id=access_key_id,
+            secret_access_key=secret_access_key,
+            session_token=session_token,
+        )
     )
+    mock_artifact_repo = mock.MagicMock(autospec=S3ArtifactRepository)
+    with mock.patch("mlflow.utils.rest_utils.http_request") as request_mock, mock.patch(
+        "mlflow.store.artifact.s3_artifact_repo.S3ArtifactRepository"
+    ) as s3_artifact_repo_class_mock:
+        s3_artifact_repo_class_mock.return_value = mock_artifact_repo
+        storage_location = "s3://blah"
+        source = str(tmp_path)
+        model_name = "model_1"
+        version = str(1)
+        request_mock.side_effect = get_request_mock(
```

Comment:
```python
with mock.patch("mlflow.utils.rest_utils.http_request", side_effect=get_request_mock(...))
```

We can call `mock.patch` with `side_effect`.

Reply:
Oops, thanks :) I keep forgetting